deltachat-ffi/deltachat.h

```diff
@@ -463,6 +463,64 @@ char*           dc_get_info                  (const dc_context_t* context);
 char*           dc_get_oauth2_url            (dc_context_t* context, const char* addr, const char* redirect_uri);
 
 
+#define DC_CONNECTIVITY_NOT_CONNECTED        1000
+#define DC_CONNECTIVITY_CONNECTING           2000
+#define DC_CONNECTIVITY_WORKING              3000
+#define DC_CONNECTIVITY_INTERRUPTING_IDLE    4000
+#define DC_CONNECTIVITY_CONNECTED            5000
+
+
+/**
+ * Get the current connectivity, i.e. whether the device is connected to the IMAP server.
+ * One of:
+ * - DC_CONNECTIVITY_NOT_CONNECTED (1000-1999): Show e.g. the string "Not connected" or a red dot
+ * - DC_CONNECTIVITY_CONNECTING (2000-2999): Show e.g. the string "Connecting…" or a yellow dot
+ * - DC_CONNECTIVITY_WORKING (3000-3999): Show e.g. the string "Getting new messages" or a spinning wheel
+ * - DC_CONNECTIVITY_INTERRUPTING_IDLE or DC_CONNECTIVITY_CONNECTED (>=4000): Show e.g. the string "Connected" or a green dot
+ *
+ * We don't use exact values but ranges here so that we can split up
+ * states into multiple states in the future.
+ *
+ * Meant as a rough overview that can be shown 
+ * e.g. in the title of the main screen.
+ *
+ * Also, you can use this to find out when the core is completely done with fetching:
+ * - call dc_start_io() (in case IO was not running)
+ * - call dc_maybe_network()
+ * - wait until the connectivity is DC_CONNECTIVITY_CONNECTED (>=5000)
+ * That's also the reason why we have DC_CONNECTIVITY_INTERRUPTING_IDLE:
+ * When you call dc_maybe_network, the connectivity is immediately set to
+ * DC_CONNECTIVITY_INTERRUPTING_IDLE (without sending DC_EVENT_CONNECTIVITY_CHANGED).
+ * This way, you can be sure that as soon as the connectivity is DC_CONNECTIVITY_CONNECTED
+ * again, all emails were fetched.
+ *
+ * If the connectivity changes, a DC_EVENT_CONNECTIVITY_CHANGED will be emitted.
+ *
+ * @memberof dc_context_t
+ * @param context The context object.
+ * @return The current connectivity.
+ */
+int             dc_get_connectivity          (dc_context_t* context);
+
+
+/**
+ * Get an overview over the current connectivity, and possibly more statistics.
+ * Meant to give the user more insight about the current status than
+ * the basic connectivity info returned by dc_get_connectivity(); show this
+ * e.g., if the user taps on said basic connectivity info.
+ *
+ * If this page changes, a DC_EVENT_CONNECTIVITY_CHANGED will be emitted.
```

Comment:
Is it always true? If `dc_maybe_network` is called, will the page contents change without emitting an event?

Reply:
> If `dc_maybe_network` is called, will the page contents change without emitting an event?

No. `dc_maybe_network` only changes the connectivity so that the UI can be sure that after calling dc_maybe_network() it is enough to wait for CONNECTED and then can be sure that things were fetched once. Everything user-facing does not distinguish between CONNECTED and INTERRUPTING_IDLE.